### PR TITLE
ci: add lint-extras github workflow

### PR DESCRIPTION
**Pull request by riya-singhal31**
add lint-extras github workflow

### CODE REVIEW

File: .github/workflows/lint-extras.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+permissions:
+  contents: read
+
+jobs:
+  lint-extras:
+    name: lint-extras
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: lint-extras
```
Comment (nixpanic) [inline .github/workflows/lint-extras.yaml:19]: there is no linter action here, so the job only checksout the git repository, nothing more

Maybe you want to use super-linter like is done in https://github.com/csi-addons/spec/blob/main/.github/workflows/linter.yaml ?

**Comment by nixpanic** [comment]
#12 is an open issue for this. Depending on what you include in the linter CI job, that issue can be closed with this PR.

**Comment by nixpanic** [comment]
Also see  #368 for the yaml-lint job.

File: .github/workflows/lint-extras.yaml (hunk 1 above)
Comment (riya-singhal31) [reply .github/workflows/lint-extras.yaml]: Right Niels, Thank you! 
Updated the PR.

**Comment by nixpanic** [comment]
The super-linter is running for 1+ hour now. Many complaints are about files under `vendor/` and `tools/vendor/`, these directories can be excluded from testing.

**Comment by riya-singhal31** [comment]
> The super-linter is running for 1+ hour now. Many complaints are about files under `vendor/` and `tools/vendor/`, these directories can be excluded from testing.

Yes, I too looked at this. Will be updating the PR with this. Thanks! :)

File: .github/workflows/lint-extras.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  test-lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: checkout source
+        uses: actions/checkout@v3
+
+      - name: Lint Code Base
+        uses: github/super-linter@v5
+        env:
```
Comment (nixpanic) [inline .github/workflows/lint-extras.yaml:23]: disable yamllint, as that runs in its own workflow?

```
VALIDATE_YAML: false
```

**Comment by riya-singhal31** [comment]
Update- have disabled few linters, will create separate PR for their fix enabling them.

**Comment by riya-singhal31** [comment]
Update will be done in https://github.com/csi-addons/kubernetes-csi-addons/issues/405

File: .github/workflows/lint-extras.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          VALIDATE_GO: false
+          VALIDATE_JSCPD: false
+          VALIDATE_KUBERNETES_KUBECONFORM: false
+          VALIDATE_MARKDOWN: false
+          VALIDATE_PROTOBUF: false
+<<<<<<< Updated upstream
+          
+=======
+          VALIDATE_YAML: false
+>>>>>>> Stashed changes
```
Comment (nixpanic) [inline .github/workflows/lint-extras.yaml]: this should be cleaned up from the 1st commit in the PR

File: .github/workflows/lint-extras.yaml [hunk 4]
```diff
@@ -29,8 +29,4 @@ jobs:
           VALIDATE_KUBERNETES_KUBECONFORM: false
           VALIDATE_MARKDOWN: false
           VALIDATE_PROTOBUF: false
```
Comment (nixpanic) [inline .github/workflows/lint-extras.yaml:31]: the 2nd commit shold not contain any changes to the workflow file

File: .github/workflows/lint-extras.yaml (hunk 3 above)
Comment (riya-singhal31) [reply .github/workflows/lint-extras.yaml]: Thanks Niels, Updated.
PTAL

**Review by nixpanic — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

**Comment by Rakshith-R** [comment]
@mergifyio rebase

**Pull request merged**